### PR TITLE
[FEAT] add i18n spanish, french and english

### PR DESCRIPTION
**Pull request by sanchis**
A simple POC of translation app with [ngx-translate](https://github.com/ngx-translate).
Fixes: #52 
Add i18n for Spanish, English and French.

The translation is made with JSON files by sections and by key and values.
For example, for component side-bar.

Feel free to upload the rest of the translations.

For the moment only applied in side-bar.

- [x] translate all the app
    - [x] side-bar 
        - [x] ES
        - [x] FR
        - [x] EN
        - [x] AR
        - [x] RU  
- [ ] Find a good place to put the default language for the app for the moment is hardcoded in the file [locale.service.ts:23](https://github.com/CMEPW/Smersh/compare/master...sanchis:fix/52-i18n?expand=1#diff-3584286d31cd60b9a7f823828c2a493f3133fd7f36dab0afcd9be34cf913b259R23) [app.module.ts:160](https://github.com/CMEPW/Smersh/compare/master...sanchis:fix/52-i18n?expand=1#diff-6740fe9cf7cb28418fb392361e04d327dc5f1f814e733e7ae46a1c936bdacc7eR160)

### CODE REVIEW

**Comment by jenaye** [comment]
don't hesitate to tell me if I say something wrong in Spanish @sanchis  :)

File: client/src/app/components/add-vulns-to-host-external/add-vulns-to-host-external.component.ts [hunk 1]
```diff
@@ -15,13 +15,17 @@ import { HostModelApplication } from 'src/app/model/Host';
...
+import { LangChangeEvent, TranslateService } from '@ngx-translate/core';
+import { Language, LocaleService } from 'src/app/services/locale.service';
 
 @Component({
   selector: 'app-add-vulns-to-host-external',
   templateUrl: './add-vulns-to-host-external.component.html',
   styleUrls: ['./add-vulns-to-host-external.component.scss'],
 })
 export class AddVulnsToHostExternalComponent implements OnInit {
+  public languages = Object.keys(Language).map((lang) => Language[lang]);
```
Comment (sanchis) [inline client/src/app/components/add-vulns-to-host-external/add-vulns-to-host-external.component.ts:27]: For this file you don't need to inject or use nothing for translation library.
So revert changes from this particular component (add-vulns-to-external.component.ts)

File: client/src/app/components/client-edit/client-edit.component.html [hunk 2]
```diff
@@ -1,19 +1,19 @@
-<h1 xmlns="http://www.w3.org/1999/html">Edit client {{ id }}</h1>
+<h1 xmlns="http://www.w3.org/1999/html">{{ 'client-create.create' | translate }} {{ id }}</h1>
```
Comment (sanchis) [inline client/src/app/components/client-edit/client-edit.component.html:1]: Is correct translation ? For `client-create.create` to translate 'Edit client'?

File: client/src/app/components/mission-my/mission-my.component.ts [hunk 3]
```diff
@@ -11,11 +14,21 @@ import {MissionRouter} from "src/app/router/MissionRouter";
 export class MissionMyComponent implements OnInit {
   public missions = [];
   public roles = [];
+  public languages = Object.keys(Language).map((lang) => Language[lang]);
+  public currentLang: Language = new Locale().get() as Language;
 
-  constructor(private usersServices: UsersService, private router: Router) {}
+  constructor(
```
Comment (sanchis) [inline client/src/app/components/mission-my/mission-my.component.ts:20]: Same as add-vulns-to-host-external.component.ts you don't need to import or inject anything in your component.

Comment (sanchis) [reply client/src/app/components/mission-my/mission-my.component.ts:20]: You can revert the changes for this file

File: client/src/assets/i18n/es.json [hunk 4]
```diff
@@ -7,5 +7,41 @@
...
+        "linked": "Misiones asignadas",
+        "noMission": "No tienes ninguna misión por el momentoé",
+        "myMission": "Mis missions",
+        "myRoles": "Mis rôles"
+    },
+    "button": {
+        "show": "Muestra",
+        "edit": "Editar",
+        "delete": "Eliminar",
+        "add": "añadir",
```
Comment (sanchis) [inline client/src/assets/i18n/es.json:22]: Capitalize first word?

File: client/src/assets/i18n/es.json [hunk 5]
```diff
@@ -7,5 +7,41 @@
...
+        "submit": "presentar"
```
Comment (sanchis) [inline client/src/assets/i18n/es.json:23]: Capitalize first word?

File: client/src/assets/i18n/es.json [hunk 6]
```diff
@@ -7,5 +7,41 @@
...
+    },
+    "add-vulns-to-host-external": {
+        "addVulnTo": "Añadir vulnerabilidad a",
+        "selectVulns": "Seleccione la vulnerabilidad",
+        "selectImpacts": "Seleccione el impacto",
+        "currentState": "Estado actual",
+        "createNewVuln": "Crear nuevo vuln",
+        "addPicture": "insertar imagen"
```
Comment (sanchis) [inline client/src/assets/i18n/es.json:31]: Capitalize first word?

File: client/src/assets/i18n/es.json [hunk 7]
```diff
@@ -7,5 +7,41 @@
         "MISSIONS": "Misiones",
         "USERS": "Usuarios",
         "VULNS":"Vulnerabilidades"
+    },
+    "dashboard": {
+        "welcome": "Bienvenido a su tablero de mandos",
+        "linked": "Misiones asignadas",
+        "noMission": "No tienes ninguna misión por el momentoé",
+        "myMission": "Mis missions",
+        "myRoles": "Mis rôles"
```
Comment (sanchis) [inline client/src/assets/i18n/es.json:16]: To `Mis roles`

File: client/src/assets/i18n/es.json [hunk 8]
```diff
@@ -7,5 +7,41 @@
         "MISSIONS": "Misiones",
         "USERS": "Usuarios",
         "VULNS":"Vulnerabilidades"
+    },
+    "dashboard": {
+        "welcome": "Bienvenido a su tablero de mandos",
+        "linked": "Misiones asignadas",
+        "noMission": "No tienes ninguna misión por el momentoé",
+        "myMission": "Mis missions",
```
Comment (sanchis) [inline client/src/assets/i18n/es.json:15]: To `Mis misiones`

File: client/src/assets/i18n/es.json [hunk 9]
```diff
@@ -7,5 +7,41 @@
         "MISSIONS": "Misiones",
         "USERS": "Usuarios",
         "VULNS":"Vulnerabilidades"
+    },
+    "dashboard": {
+        "welcome": "Bienvenido a su tablero de mandos",
+        "linked": "Misiones asignadas",
+        "noMission": "No tienes ninguna misión por el momentoé",
```
Comment (sanchis) [inline client/src/assets/i18n/es.json:14]: To `No tienes ninguna misión por el momento`

File: client/src/assets/i18n/es.json [hunk 10]
```diff
@@ -7,5 +7,41 @@
...
+    "client-create": {
+        "create": "Crear un nuevo cliente",
+        "name": "Nombre cliente",
+        "firstname": "Nombre",
+        "lastname": "Apellido",
+        "phone": "Teléfono"
+    },
+    "edit-vuln-with-state": {
+        "vuln": "Vulnerabilidad",
+        "target": "Target"
```
Comment (sanchis) [inline client/src/assets/i18n/es.json:42]: To `Objetivo`

File: client/src/assets/i18n/es.json (hunk 5 above)
Comment (sanchis) [reply client/src/assets/i18n/es.json:23]: I think the correct translation for submit is `Enviar`

File: client/src/assets/i18n/en.json [hunk 11]
```diff
@@ -7,5 +7,41 @@
...
+    },
+    "client-create": {
+        "create": "Create a new client",
+        "name": "Client name",
+        "firstname": "Firstname",
+        "lastname": "Lastname",
+        "phone": "phone"
+    },
+    "edit-vuln-with-state": {
+    "vuln": "Vulnerability",
```
Comment (sanchis) [inline client/src/assets/i18n/en.json:41]: Tab to correct format.

File: client/src/app/components/client-edit/client-edit.component.html (hunk 2 above)
Comment (jenaye) [reply client/src/app/components/client-edit/client-edit.component.html:1]: its the same field so i took client-create key, i probably should create a client-edit

**Comment by jenaye** [comment]
image

still in WIP

**Comment by jenaye** [comment]
i need to fix locale for edit vuln and add vulns and it will be ok

**Comment by jenaye** [comment]
image

image

**Pull request merged**